src/DIRAC/FrameworkSystem/API/AuthHandler.py

```diff
@@ -435,9 +435,9 @@ def __researchDIRACGroup(self, extSession, chooseScope, state):
         username = extSession["authed"]["username"]
         # Requested arguments in first request
         provider = firstRequest.provider
-        self.log.debug(f"Next groups has been found for {username}:", ", ".join(firstRequest.groups))
+        self.log.debug("The following groups found", f"for {username}: {', '.join(firstRequest.groups)}")
 
-        # Researche Group
+        # Research Group
```

Comment:
"Research Group" makes no sense as a comment.
What is the difference between `firstRequest.groups` and the groups we look up now?

Naively I would have expected we check `getGroupsForUser` against `firstRequest.groups` but that doesn't happen in any obvious way. And then we just return `firstRequest.groups` if that is true'ish

Reply:
User is authorized in two steps. First, through the auth flow with an external provider, e.g. WLCG_IAM, and then by DIRAC as auth server. User can from the beginning specify the desired group, e.g. with -g option of the dirac-login command. Then this group will be known as part of the first request. If the group is not specified, the user is given a choice in the second auth flow via the DIRAC auth server where the user will presented a list of valid groups to choose from.

Reply:
I have modified comments slightly to make it more explicit